### ISSUE TITLE
[Question] Run recorder inside a Iframe

### ISSUE THREAD

**Issue by omkarK06** [ruifigueira/playwright-crx#20]
Question:
Recorder-crx records user interactions from that tab as we attach the tab id to the crx app. 
But is it possible to attach iframe as document from the tab and record interactions only from the iframe.

**Comment by ruifigueira** [comment]
Yes, as long as the iframe belongs to an attached page, it should record actions in that iframe.

Did you have problems with nested iframes not being recorded?

**Comment by omkarK06** [comment]
Yes faced problems in this case, where the main web application tab to which the recorder is attached and the iframe content are sourced from different domains or systems.

**My end goal**
The iframe document (a different domain or website rendered in iframe) is embedded within the web application, and I am specifically interested in recording the user interactions within this iframe only, not the surrounding web application.
The goal is to exclusively capture and generate test scripts for events that occur within the iframe's document.

I would be happy to discuss this use case and contribute to this project.

**Comment by ruifigueira** [comment]
I understand. I tried it here and did replicate the issue with the following test:

```js
test("should work in cross-domain iframes", async ({ runCrxTest }) => {
  await runCrxTest(async ({ page }) => {
    await page.goto('https://www.srimax.com/cross-domain-iframe-using-html5/');
    await page.frameLocator('iframe[title="Twitter Follow Button"]').getByRole('link', { name: 'Follow' }).click();
    await page.waitForEvent('popup');
  });
});
```

Let me investigate it and get back to you.

**Comment by ruifigueira** [comment]
It seems to be some limitation of [Page.getFrameTree](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-getFrameTree) on [chrome.debugger](https://developer.chrome.com/docs/extensions/reference/api/debugger).

In native playwright, the same `Page.getFrameTree` call returns the full frame tree (cross-domain iframes included), while using `chrome.debugger` it doesn't return cross-domain frames.

I'll probably open an issue in chromium for that.

**Comment by ruifigueira** [comment]
upstream bug: https://issues.chromium.org/issues/333981074

**Comment by ruifigueira** [comment]
Actually, it is not working, because I have to change a few things. Hopefully in the next version this will be working!

**Comment by ruifigueira** [comment]
@omkarK06 I just released version 0.6.0 which already supports out-of-process iframes, but only for chrome 126+, which is expected to be stable next month.
